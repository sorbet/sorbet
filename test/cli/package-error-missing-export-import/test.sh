cd test/cli/package-error-missing-export-import || exit 1

../../../main/sorbet --silence-dev-message --stripe-packages \
  --stripe-packages-hint-message="Try running generate-packages.sh" \
  --max-threads=0 other missing_import 2>&1

../../../main/sorbet --silence-dev-message --stripe-packages \
  --stripe-packages-hint-message="Try running generate-packages.sh" \
  --max-threads=0 other imported_as_test 2>&1

../../../main/sorbet --silence-dev-message --stripe-packages \
  --stripe-packages-hint-message="Try running generate-packages.sh" \
  --max-threads=0 other imported_as_test_unit 2>&1
