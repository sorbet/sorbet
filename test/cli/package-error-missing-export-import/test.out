missing_import/foo_class.rb:5: `Other::OtherClass` resolves but is not exported from `Other` and `Other` is not imported https://srb.help/3718
     5 |    Other::OtherClass
            ^^^^^^^^^^^^^^^^^
    other/other_class.rb:4: Defined here
     4 |  class OtherClass
          ^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    missing_import/__package.rb:3: Insert `import Other`
     3 |class Foo::MissingImport < PackageSpec
                                              ^
    other/__package.rb:3: Insert `export Other::OtherClass`
     3 |class Other < PackageSpec
                                 ^
  Note:
    Try running generate-packages.sh
Errors: 1
imported_as_test/foo_class.rb:5: `Other::OtherClass` resolves but is not exported from `Other` and `Other` is `test_import`ed https://srb.help/3720
     5 |    Other::OtherClass
            ^^^^^^^^^^^^^^^^^
    other/other_class.rb:4: Defined here
     4 |  class OtherClass
          ^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    imported_as_test/__package.rb:3: Insert `import Other`
     3 |class Foo::MissingImport < PackageSpec
                                              ^
    imported_as_test/__package.rb:4: Delete
     4 |  test_import Other
        ^^^^^^^^^^^^^^^^^^^
    other/__package.rb:3: Insert `export Other::OtherClass`
     3 |class Other < PackageSpec
                                 ^
  Note:
    Try running generate-packages.sh
Errors: 1
imported_as_test_unit/test/foo_class.rb:5: `Other::OtherClass` resolves but is not exported from `Other` and `Other` is `test_import`ed for only .test.rb files https://srb.help/3720
     5 |    Other::OtherClass
            ^^^^^^^^^^^^^^^^^
  Note:
    This is because this `test_import` is declared with `only: 'test_rb'`, which means the constant can only be used in `.test.rb` files.
    other/other_class.rb:4: Defined here
     4 |  class OtherClass
          ^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    imported_as_test_unit/__package.rb:3: Insert `test_import Other`
     3 |class Foo::MissingImport < PackageSpec
                                              ^
    imported_as_test_unit/__package.rb:4: Delete
     4 |  test_import Other, only: "test_rb"
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    other/__package.rb:3: Insert `export Other::OtherClass`
     3 |class Other < PackageSpec
                                 ^
  Note:
    Try running generate-packages.sh
Errors: 1
